spec.md

```diff
@@ -49,6 +49,119 @@ The following will not be part of the specification:
+     producer-assigned classification.
+   - identifying the temporal order of the events relative to the
+     originating context and/or relative to a wall-clock.
+   - understanding the context-related detail information carried
+     in the event.
+
+   In some cases, the consuming application might be interested in:
+   - obtaining further details about the event's subject from the
+     originating context, like obtaining detail information about a
+     changed object that requires privileged access authorization.
+   - interact with the event's subject at the originating context,
+     for instance reading a storage blob after having been informed
+     that this blob has just been created.
+
+   Consumer interests motivate requirements for which information
+   producers ought to include an event.
+
+3) Middleware routes events from producers to consumers, or onwards
+   to other middleware. Applications producing events might delegate
+   certain tasks arising from their consumers' requirements to
+   middleware:
+
+   - Management of many concurrent interested consumers for one of
+     multiple classes or originating contexts of events
+   - Processing of filter conditions over a class or originating context
+     of events on behalf of consumers.
+   - Transcoding, like encoding in MsgPack after decoding from JSON
+   - Transformation that changes the event's structure, like mapping from
+     a proprietary format to Cloud Events, while preserving the
+     identity and semantic integrity of the event.
+   - Instant "push-style" delivery to interested consumers.
+   - Storing events for eventual delivery, either for pick-up initiated
+     by the consumer ("pull"), or initiated by the middleware ("push")
+     after a delay.
+   - Observing event content or event flow for monitoring or
+     diagnostics purposes.
+
+   To satisfy these needs, middleware will be interested in:
+   - A metadata discriminator usable for classification of events so that
+     consumers can express interest in one or multiple such classes.
+   - A metadata discriminator that allows distinguishing the subject of
+     a particular event of that class.
+   - An indicator for the encoding of the event and its data.
+   - An indicator for the structural layout (schema) for the event and
+     its data.
+
+   Whether its events are available for consumption via a middleware is
+   a delegation choice of the producer.
+
+4) Frameworks and other abstractions make interactions with event platform
+   infrastructure simpler, and often provide common API surface areas
+   for multiple event platform infrastructures.
+
+   Frameworks are often used for turning events into an object graph,
+   and to dispatch the event to some specific handling user-code or
+   user-rule that permits the consuming application to react to
+   a particular kind of occurrence in the originating context and
+   on a particular subject.
```

Comment:
nit: user => application developer

Reply:
I'm telling an app to specifically show the MSFT, GOOG, and IBM stock ticks for me, it's arguable that the subscription rule has been user supplied. This section is about frameworks that deal with dispatching events and therefore I chose this to be a bit broader than just application developers.